APTITUDE/APTITUDETEST.Data/Data/QuestionRepository.cs

```diff
@@ -731,6 +749,51 @@ private bool ValidateTopics(ImportQuestionFieldsVM question)
             return true;
         }
 
+        private string GenerateSequence(int duplicateId)
+        {
+            if (duplicateId == 0)
+            {
+                int highestNumber = getHighestSequence();
+                return (highestNumber + 1).ToString();
+            }
+            else
+            {
+                string temp = _context.Questions.Where(q => q.ParentId == duplicateId && q.IsDeleted != true).Select(x => x.Sequence).ToList()
+               .OrderByDescending(s => s)
+               .FirstOrDefault();
+                if (temp == null)
+                {
+                    return _context.Questions.Where(q => q.Id == duplicateId).Select(q => q.Sequence).FirstOrDefault() + ((char)SequenceStart.Start).ToString();
+                }
+                else
+                {
+                    string num = new string(temp.TakeWhile(char.IsDigit).ToArray());
+                    char alphabet = new string(temp.SkipWhile(char.IsDigit).ToArray())[0];
+                    char nextChar = (char)(alphabet + 1);
+                    return num + nextChar;
+                }
+            }
+        }
+
+        private void FillImportQuestionSequence(List<ImportQuestionFieldsVM> questions)
+        {
+            int start = getHighestSequence() + 1;
+            questions.ForEach(q =>
+            {
+                q.sequence = start.ToString();
+                start++;
+            });
+        }
+
+        private int getHighestSequence()
+        {
+            return _context.Questions.Select(x => x.Sequence).ToList()
```

Comment:
check here - ToList() really required ?

Reply:
List is required as we are getting list of sequence and then ordering it